lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -286,5 +289,101 @@ def with_truncated_backtrace(child)
         end
       end
     end
+
+    # Provides a single exception instance that provides access to
+    # multiple sub-exceptions. This is used in situations where a single
+    # individual spec has multiple exceptions, such as one in the `it` block
+    # and one in an `after` block.
+    class MultipleExceptionError < StandardError
+      # @private
+      # Used so there is a common module in the ancestor chain of this class
+      # and `RSpec::Expectations::MultipleExpectationsNotMetError`, which allows
+      # code to detect exceptions that are instances of either, without first
+      # checking to see if rspec-expectations is loaded.
+      module InterfaceTag
+        # Appends the provided exception to the list.
+        # @param exception [Exception] Exception to append to the list.
+        # @private
+        def add(exception)
+          # `PendingExampleFixedError` can be assigned to an example that initially has no
+          # failures, but when the `aggregate_failures` around hook completes, it notifies of
+          # a failure. If we do not ignore `PendingExampleFixedError` it would be surfaced to
+          # the user as part of a multiple exception error, which is undesirable. While it's
+          # pretty weird we handle this here, it's the best solution I've been able to come
+          # up with, and `PendingExampleFixedError` always represents the _lack_ of any exception
+          # so clearly when we are transitioning to a `MultipleExceptionError`, it makes sense to
+          # ignore it.
+          return if Pending::PendingExampleFixedError === exception
+
+          all_exceptions << exception
+
+          if exception.class.name =~ /RSpec/
```

Comment:
Yeah, it is gross :(.  We already treat exceptions with "RSpec" in the class name as special here:

https://github.com/rspec/rspec-core/blob/bebc52c9e7c411d4b1951e7b3169eeb8528daab8/lib/rspec/core/formatters/exception_presenter.rb#L94

I was merely mirroring that.